### PR TITLE
selinuxpolicy: Consume updated deployer with selinux policy updates

### PR DESCRIPTION
**Pull request by swatisehgal**
We need different selinux policies based on the platform version.
Recently, changes were implemented in the deployer which can be
vendored in here.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
> Recently, changes were implemented in the deployer which can be vendored in here.

Good.
Better: "starting with version X.Y AND/OR commit ABCDEF, changes were..."

**Comment by swatisehgal** [comment]
Additional updates are needed in the controllers and hack directory. Working on that.

File: hack/lsplatform/lsplatform.go [hunk 1]
```diff
@@ -12,17 +12,21 @@ import (
 
 func main() {
 	platformName := ""
+	platformVersion := ""
 	flag.StringVar(&platformName, "is-platform", "", "check if the detected platform matches the given one")
+	flag.StringVar(&platformVersion, "is-platform-version", "", "check if the detected platform version matches the given one")
```
Comment (fromanirh) [inline hack/lsplatform/lsplatform.go:17]: version checking is not yet needed for this tool, we can elide this change from this PR

File: main.go [hunk 2]
```diff
@@ -149,7 +158,7 @@ func main() {
 		namespace = defaultNamespace
 	}
 
-	rteManifests, err := rtemanifests.GetManifests(clusterPlatform, namespace)
+	rteManifests, err := rtemanifests.GetManifests(clusterPlatform, userPlatformVersion, namespace)
```
Comment (fromanirh) [inline main.go:161]: I think we should use `clusterVersion` instead of `userPlatformVersion`

**Review by fromanirh — COMMENTED** [review]
partial review

File: hack/lsplatform/lsplatform.go (hunk 1 above)
Comment (swatisehgal) [reply hack/lsplatform/lsplatform.go:17]: Okay, will remove.

File: controllers/numaresourcesoperator_controller_test.go [hunk 3]
```diff
@@ -56,18 +56,19 @@ import (
 )
 
 const (
-	testImageSpec = "quay.io/openshift-kni/numaresources-operator:ci-test"
+	testImageSpec     = "quay.io/openshift-kni/numaresources-operator:ci-test"
+	defaultOCPVersion = "v4.11"
```
Comment (fromanirh) [inline controllers/numaresourcesoperator_controller_test.go:60]: I'm trying to think if using `v4.10` would make any difference. I guess we can keep 4.11.

File: test/utils/configuration/configuration.go [hunk 4]
```diff
@@ -58,14 +61,28 @@ func init() {
 		panic(fmt.Errorf("failed to parse machine config pool update interval: %w", err))
 	}
 
-	Platform, err = detect.Detect()
-	if err != nil {
+	platDetect, reason := detect.FindPlatform(platform.Platform("Unknown"))
```
Comment (fromanirh) [inline test/utils/configuration/configuration.go:64]: in this case we do want to try autodetecting, so I'd use `detect.Platform()`
additionally, `Platform` is a really bad name for a variable (my fault) so let's take this chance to rename it to anything else which doesn't clash with the `platform.Platform` type.

File: test/utils/configuration/configuration.go [hunk 5]
```diff
@@ -58,14 +61,28 @@ func init() {
...
+	if Platform == platform.Unknown {
 		Platform = getPlatformFromEnv(envVarPlatform)
 	}
+
 	if Platform == platform.Unknown {
 		Platform = platform.OpenShift
 		klog.Infof("forced to %q: failed to detect a platform: %w", Platform, err)
 	}
+
+	versionDetect, reason := detect.FindVersion(Platform, platform.MissingVersion)
```
Comment (fromanirh) [inline test/utils/configuration/configuration.go:76]: same, we can simplify like
```
	platVer, err = detect.Version(plat)
	if err != nil {
		platVer = getPlatformVersionFromEnv(envVarPlatformVersion)
	}
	if platVer == platform.MissingVersion {
		...
```

File: test/utils/configuration/configuration.go [hunk 6]
```diff
@@ -82,6 +99,18 @@ func getPlatformFromEnv(envVar string) platform.Platform {
 	return platform.Unknown
 }
 
+func getPlatformVersionFromEnv(envVar string) (platform.Version, error) {
```
Comment (fromanirh) [inline test/utils/configuration/configuration.go:102]: nice!

File: test/utils/configuration/configuration.go (hunk 4 above)
Comment (swatisehgal) [reply test/utils/configuration/configuration.go]: Ack, fixed.

File: test/utils/configuration/configuration.go (hunk 5 above)
Comment (swatisehgal) [reply test/utils/configuration/configuration.go]: Ack, Thanks. Fixed.

File: test/utils/configuration/configuration.go (hunk 6 above)
Comment (swatisehgal) [reply test/utils/configuration/configuration.go:101]: :)

File: controllers/numaresourcesoperator_controller_test.go (hunk 3 above)
Comment (swatisehgal) [reply controllers/numaresourcesoperator_controller_test.go:60]: Yeah, it does not make a significant difference the version we use. I had to provide a value and thought its best to use the latest version here.

Comment (fromanirh) [reply controllers/numaresourcesoperator_controller_test.go:60]: that's fine indeed!

File: hack/lsplatform/lsplatform.go [hunk 7]
```diff
@@ -15,14 +15,16 @@ func main() {
 	flag.StringVar(&platformName, "is-platform", "", "check if the detected platform matches the given one")
 	flag.Parse()
 
-	clusterPlatform, err := detect.Detect()
-	if err != nil {
-		log.Fatalf("error detecting the platform: %v", err)
+	plat, reason := detect.FindPlatform(platform.Platform(platformName))
```
Comment (fromanirh) [inline hack/lsplatform/lsplatform.go:18]: still not sure about this. I'll check.

**Review by fromanirh — COMMENTED** [review]
let me just check `lsplatform` and we can merge

File: hack/lsplatform/lsplatform.go (hunk 7 above)
Comment (swatisehgal) [reply hack/lsplatform/lsplatform.go]: Fixed.

**Review by fromanirh — COMMENTED** [review]
/lgtm
thanks, now looks good. Kudos!

**Comment by fromanirh** [comment]
ok, errors are not a glitch and need to be checked. Most likely unrelated to this PR

**Comment by fromanirh** [comment]
because #324 is progressing past the point this test was failing

**Comment by fromanirh** [comment]
we need extra fixes, added in #338 (preserving the base commits here)

**Pull request merged**